x-pack/plugin/autoscaling/src/internalClusterTest/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageIT.java

```diff
@@ -392,15 +393,20 @@ public void testScaleDuringSplitOrClone() throws Exception {
         setTotalSpace(dataNode1Name, enoughSpace);
         setTotalSpace(dataNode2Name, enoughSpace);
 
-        // validate initial state looks good
-        GetAutoscalingCapacityAction.Response response = capacity();
-        assertThat(response.results().keySet(), equalTo(Set.of(policyName)));
-        assertThat(response.results().get(policyName).currentCapacity().total().storage().getBytes(), equalTo(enoughSpace * 2));
-        assertThat(response.results().get(policyName).requiredCapacity().total().storage().getBytes(), equalTo(enoughSpace * 2));
-        assertThat(
-            response.results().get(policyName).requiredCapacity().node().storage().getBytes(),
-            equalTo(used + LOW_WATERMARK_BYTES + ReactiveStorageDeciderService.NODE_DISK_OVERHEAD)
-        );
+        // It might take a while until the autoscaling polls the node information of dataNode2 and
+        // provides a complete autoscaling capacity response
+        assertBusy(() -> {
+            // validate initial state looks good
+            GetAutoscalingCapacityAction.Response response = capacity();
```

Comment:
If the NPE is just because there is no result in the response, why not only assertBusy getting the capacity() call, and leave the rest out of it?

Reply:
More of a question/nit! :-)

Reply:
Yes, that makes sense, it just felt like the entire assertion block belonged together, but it would make the test slower if there's some other failure. I'll amend it 👍